src/Microsoft.Azure.WebJobs.Host/Blobs/Listeners/ScanBlobScanLogHybridPollingStrategy.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Listeners;
+using Microsoft.Azure.WebJobs.Host.Storage;
+using Microsoft.Azure.WebJobs.Host.Storage.Blob;
+using Microsoft.Azure.WebJobs.Host.Timers;
+using Microsoft.WindowsAzure.Storage;
+
+namespace Microsoft.Azure.WebJobs.Host.Blobs.Listeners
+{
+    internal sealed class ScanBlobScanLogHybridPollingStrategy : IBlobListenerStrategy
+    {
+        private static readonly TimeSpan TwoSeconds = TimeSpan.FromSeconds(2);
+        private enum PollStrategy { ScanBlobs, ScanLogs, NotDetermined };
+        private int _scanBlobThreshold = 5000;
+        private readonly IDictionary<IStorageBlobContainer, PollStrategy> _pollStrategies;
+        private readonly IDictionary<IStorageBlobContainer, ICollection<ITriggerExecutor<IStorageBlob>>> _registrations;
+        private ScanContainersStrategy _scanContainerStrategy;
```

Comment:
This existing ScanContainersStrategy is basically untested code, since it effectively dead code since it only runs when dev storage is being used (which we don't support). So if we're now going to use it in the default strategy we'll need to test it thoroughly.